src/zenml/integrations/kubernetes/flavors/kubernetes_orchestrator_flavor.py

```diff
@@ -36,12 +36,16 @@ class KubernetesOrchestratorSettings(BaseSettings):
             block until all steps finished running on Kubernetes.
         timeout: How many seconds to wait for synchronous runs. `0` means
             to wait for an unlimited duration.
+        service_account_name: Name of the service account to use for the
+            orchestrator pod. If not provided, a new service account with "edit"
+            permissions will be created.
         pod_settings: Pod settings to apply.
     """
 
     synchronous: bool = False
     timeout: int = 0
 
+    service_account_name: Optional[str] = None
```

Comment:
Is this something relevant for other Kubernetes-based orchestrators and should therefore be in the `KubernetesPodSettings` instead?

Reply:
I just checked and couldn't find any flavors that have a service account as a setting. However, the following flavors have a service account as a config value:
- `KserveModelDeployer`
- `KanikoImageBuilder`
- `VertexOrchestrator`
- `KubernetesSparkStepOperator`

Since these are all different stack component types, I don't really see a way to unify this.

I'm also not sure how exactly it works for kfp or other k8s-based orchestrators. My guess would be that you already need to have the service account set up to deploy the tool in the first place. Perhaps you or @stefannica know this better?

Reply:
Yep I just checked and it seems not possible to set this for KFP (and therefore our other kubenertes-based orchestrators), so the solution you implemented looks good 👍